bundles/org.openhab.binding.generacmobilelink/src/main/java/org/openhab/binding/generacmobilelink/internal/handler/GeneracMobileLinkAccountHandler.java

```diff
@@ -61,191 +64,305 @@
-            if (thing.getStatusInfo().getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR) {
-                // bad credentials, stop trying to login
-                stopPoll();
+
+            String id = String.valueOf(apparatus.apparatusId);
+            apparatusesCache.put(id, apparatus);
+
+            Optional<Thing> thing = getThing().getThings().stream().filter(
+                    t -> t.getConfiguration().as(GeneracMobileLinkGeneratorConfiguration.class).generatorId.equals(id))
+                    .findFirst();
+            if (!thing.isPresent()) {
+                discoveryService.generatorDiscovered(apparatus, getThing().getUID());
+            } else {
+                ThingHandler handler = thing.get().getHandler();
+                if (handler != null) {
+                    updateGeneratorThing(handler, apparatus);
+                }
             }
         }
     }
 
-    private void getStatuses(boolean retry) throws InterruptedException {
-        if (authToken == null) {
-            return;
+    private void updateGeneratorThing(ThingHandler handler, Apparatus apparatus)
+            throws IOException, SessionExpiredException {
+        ApparatusDetail detail = getEndpoint(ApparatusDetail.class, "/v1/Apparatus/details/" + apparatus.apparatusId);
+        if (detail != null) {
+            ((GeneracMobileLinkGeneratorHandler) handler).updateGeneratorStatus(apparatus, detail);
+        } else {
+            logger.debug("Could not decode apparatuses detail response");
         }
-        HTTPResult result = sendRequest(BASE_URL + "/Generator/GeneratorStatus", HttpMethod.GET, authToken, null, null);
-        if (result.responseCode == HttpStatus.OK_200) {
-            generators = gson.fromJson(result.content, GeneratorStatusResponseDTO.class);
-            updateGeneratorThings();
-            if (getThing().getStatus() != ThingStatus.ONLINE) {
-                updateStatus(ThingStatus.ONLINE);
+    }
+
+    private @Nullable <T> T getEndpoint(Class<T> clazz, String endpoint) throws IOException, SessionExpiredException {
+        try {
+            ContentResponse response = httpClient.newRequest(API_BASE + endpoint).send();
+            if (response.getStatus() == 204) {
+                // no data
+                return null;
             }
-        } else {
-            if (retry) {
-                logger.debug("Retrying status request");
-                getStatuses(false);
-            } else {
-                handleErrorResponse(result);
+            if (response.getStatus() != 200) {
+                throw new SessionExpiredException("API returned status code: " + response.getStatus());
             }
+            String data = response.getContentAsString();
+            logger.debug("getEndpoint {}", data);
+            return gson.fromJson(data, clazz);
+        } catch (InterruptedException | TimeoutException | ExecutionException | JsonSyntaxException e) {
+            throw new IOException(e);
```

Comment:
Consider if `InterruptedException` should be handled differently/separately, for example by  `Thread.currentThread().interrupt()` before throwing, or not catching it here.